src/components/editor/EditorLogin.jsx

```diff
@@ -1,73 +1,104 @@
 import BaseComponent from 'lib/BaseComponent';
 import React from 'react';
 import { browserHistory } from 'react-router';
-import { hash } from 'lib/utilities';
+import { googleClientID } from 'lib/utilities';
 import _ from 'lodash';
 
-import TextField from 'material-ui/TextField';
-import RaisedButton from 'material-ui/RaisedButton';
-
 const styles = {
-  buttons: {
-    marginTop: 12,
-    marginBottom: 24,
+  button: {
+    marginTop: 36,
   },
 };
 
+
 export default class Login extends BaseComponent {
   constructor(props) {
     super(props);
-    this.safeSetState({
-      passwordValue: '',
-    });
-    this.handlePasswordChange = this.handlePasswordChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = {
+      GoogleAPIReady: false,
+    };
+    this.handleRedirect = this.handleRedirect.bind(this);
+    this.onSignInSuccess = this.onSignInSuccess.bind(this);
   }
 
-  handlePasswordChange(e) {
-    this.safeSetState({
-      passwordValue: e.target.value,
-    });
-  }
+  componentDidMount() {
+    const interval = setInterval(() => {
+      // GoogleAPILoaded is a global var, set to true when Google platform.js script loaded
+      if (GoogleAPILoaded) { // eslint-disable-line no-undef
+        this.renderButton();
+        window.gapi.load('auth2', () => {
+          this.setState({
```

Comment:
My guess was safeSetState does not force a re-render after the state changes. I tested with adding a comment to `render()` and the page only renders once - so it won't allow the `opacity` change we add to the login button.

Reply:
Hey, I figured out why, you have to add `super.componentDidMount()` at the top of the `componentDidMount` function since you're overwriting the inherited `componentDidMount` function, and then `safeSetState` will work